### PR TITLE
Show baseline as dotted line at 1 for speedup

### PR DESCRIPTION
**Pull request by ajpal**
image

Also puts a scrollbox on the benchmarks so it's more manageable:
image

### CODE REVIEW

File: infra/nightly-resources/chart.js [hunk 1]
```diff
@@ -81,7 +84,7 @@ function parseDataForChart() {
     datasets[mode] = {
       label: mode,
       backgroundColor: COLORS[mode],
-      data: Array(benchmarks.length).fill(0),
+      data: Array(benchmarks.size).fill(0),
```
Comment (oflatt) [inline infra/nightly-resources/chart.js:87]: did you mean benchmarks.size here? Idk the difference but a cursory google search says length is right for arrays

File: infra/nightly-resources/chart.js [hunk 2]
```diff
@@ -32,7 +32,10 @@ function getValue(entry) {
     if (!baseline) {
       addWarning(`No speedup baseline for ${benchmark}`);
     }
-    return baseline.hyperfine.results[0].mean / entry.hyperfine.results[0].mean;
+    const baseV = baseline.hyperfine.results[0].mean;
+    const expV = entry.hyperfine.results[0].mean;
+    // return (baseV - expV) / expV * 100;
+    return baseV / expV;
```
Comment (oflatt) [inline infra/nightly-resources/chart.js:38]: would be great to show this formula on the screen (another PR is good)

File: infra/nightly-resources/chart.js [hunk 3]
```diff
@@ -112,11 +131,6 @@ function initializeChart() {
     type: "bar",
     data: chartData,
     options: {
-      scales: {
-        y: {
-          beginAtZero: true,
```
Comment (oflatt) [inline infra/nightly-resources/chart.js:117]: is this bc it wasn't working before?

File: infra/nightly-resources/chart.js [hunk 4]
```diff
@@ -97,6 +100,22 @@ function parseDataForChart() {
...
     });
   });
+
+  // Show baseline as dotted line at 1x if speedup
+  if (GLOBAL_DATA.chart.mode === "speedup") {
+    datasets[BASELINE_MODE] = {
+      label: BASELINE_MODE,
+      data: Array(benchmarks.size + 1).fill(1),
+      type: "line",
+      borderColor: "purple",
```
Comment (oflatt) [inline infra/nightly-resources/chart.js:110]: perhaps this should be colors[baseline]

**Review by oflatt — APPROVED** [review]
Looks great! None of my comments are blocking

File: infra/nightly-resources/chart.js (hunk 1 above)
Comment (ajpal) [reply infra/nightly-resources/chart.js:87]: Yeah, it's a set not an array

File: infra/nightly-resources/chart.js (hunk 3 above)
Comment (ajpal) [reply infra/nightly-resources/chart.js:117]: Yeah I copied some of this code from the ChartJS docs, but I don't think this was actually doing anything

**Pull request merged**